src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -0,0 +1,413 @@
+                _header._mTime = value;
+            }
+        }
+
+        /// <summary>
+        /// When the <see cref="EntryType"/> indicates an entry that can contain data, this property returns the length in bytes of such data.
+        /// </summary>
+        /// <remarks>The entry type that commonly contains data is <see cref="TarEntryType.RegularFile"/> (or <see cref="TarEntryType.V7RegularFile"/> in the <see cref="TarFormat.V7"/> format). Other uncommon entry types that can also contain data are: <see cref="TarEntryType.ContiguousFile"/>, <see cref="TarEntryType.DirectoryList"/>, <see cref="TarEntryType.MultiVolume"/> and <see cref="TarEntryType.SparseFile"/>.</remarks>
+        public long Length => _header._dataStream != null ? _header._dataStream.Length : _header._size;
+
+        /// <summary>
+        /// When the <see cref="EntryType"/> indicates a <see cref="TarEntryType.SymbolicLink"/> or a <see cref="TarEntryType.HardLink"/>, this property returns the link target path of such link.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Cannot set the link name if the entry type is not <see cref="TarEntryType.HardLink"/> or <see cref="TarEntryType.SymbolicLink"/>.</exception>
+        public string LinkName
+        {
+            get => _header._linkName;
+            set
+            {
+                if (_header._typeFlag is not TarEntryType.HardLink and not TarEntryType.SymbolicLink)
+                {
+                    throw new InvalidOperationException(SR.TarEntryHardLinkOrSymLinkExpected);
+                }
+                _header._linkName = value;
+            }
+        }
+
+        /// <summary>
+        /// Represents the Unix file permissions of the file represented by this entry.
+        /// </summary>
+        /// <remarks>The value in this field has no effect on Windows platforms.</remarks>
+        public TarFileMode Mode
+        {
+            get => (TarFileMode)_header._mode;
+            set
+            {
+                if ((int)value is < 0 or > 4095) // 4095 in decimal is 7777 in octal
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+                _header._mode = (int)value;
+            }
+        }
+
+        /// <summary>
+        /// Represents the name of the entry, which includes the relative path and the filename.
+        /// </summary>
+        public string Name
+        {
+            get => _header._name;
+            set
+            {
+                ArgumentException.ThrowIfNullOrEmpty(value);
+                _header._name = value;
+            }
+        }
+
+        /// <summary>
+        /// The ID of the user that owns the file represented by this entry.
+        /// </summary>
+        /// <remarks>This field is only supported in Unix platforms.</remarks>
+        public int Uid
+        {
+            get => _header._uid;
+            set => _header._uid = value;
```

Comment:
Any Int32 (including negative values) are valid?

Reply:
Right, needs bound exceptions.

I don't think this can be negative. Should we re-propose this as `uint` instead of `int`? cc @tannergooding

Reply:
> Should we re-propose this as uint instead of int? 

That would be my preference, but get ready for a lot of pushback at the review :)